server/models/user.js

```diff
@@ -40,6 +41,59 @@ const UserSchema = new mongoose.Schema({
   }
 });
 
+UserSchema.virtual('id').get(function () {
+  return this._id.toString();
+});
+
+UserSchema.methods = {
+  getDevice(id) {
+    return new Promise((resolve, reject) => {
+      const foundDevice = _.find(this.devices, { id });
+
+      if (foundDevice) {
+        return resolve(foundDevice);
+      }
+      const error = new APIError(`Device with ID ${id} doesn\'t exist!`, httpStatus.NOT_FOUND);
+      reject(error);
+    });
+  },
+
+  addDevice(deviceModel, index) {
```

Comment:
Yeah, and for ease of use too. If there's a default of, say, adding it to the beginning, we can just skip passing in 0 whenever we want to do that. It'd really be helpful only if there's one index we'd be using primarily.